src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalysisResult.cs

```diff
@@ -82,11 +71,81 @@ internal struct DiagnosticAnalysisResult
             IsEmpty = DocumentIds.IsEmpty && _others.IsEmpty;
         }
 
-        public DiagnosticAnalysisResult(DiagnosticAnalysisResultBuilder builder) :
-            this(builder.Project.Id, builder.Version,
-                builder.SyntaxLocals, builder.SemanticLocals, builder.NonLocals, builder.Others,
-                builder.DocumentIds, fromBuild: false)
+        public static DiagnosticAnalysisResult CreateEmpty(ProjectId projectId, VersionStamp version)
+        {
+            return new DiagnosticAnalysisResult(
```

Comment:
i find it a bit confusing that everything here is empty but `isEmpty == false`, whereas the one created by `CreateInitialResult` have all default values but `isEmpty == true`. Maybe consider changing it to `IsDefault` or something like that?